ddtrace/internal/datastreams/processor.py

```diff
@@ -342,15 +344,20 @@ def new_pathway(self, now_sec=None):
         ctx = DataStreamsCtx(self, 0, now_sec, now_sec)
         return ctx
 
-    def set_checkpoint(self, tags, now_sec=None):
+    def set_checkpoint(self, tags, now_sec=None, payload_size=0):
         if not now_sec:
             now_sec = time.time()
         if hasattr(self._current_context, "value"):
             ctx = self._current_context.value
         else:
             ctx = self.new_pathway()
             self._current_context.value = ctx
-        ctx.set_checkpoint(tags, now_sec=now_sec)
+        if "direction:out" in tags:
```

Comment:
Assuming that `tags` here means "span tags", this appears to be a coupling between the datastreams and tracing products. Storing the directionality of the message could be a good use case for `ddtrace.internal.core`.

Reply:
in this case "tags" is separate - its a dependency on the tags for custom metrics being generated.  I will change the name to clarify!